Block_call (Clean up version of PR pull/26.)

linearregression opened the pull request:
Clean up, squashed version of https://github.com/priestjim/gen_rpc/pull/26
Trial implementation of block_call.
Implementation:
* block_call(Node, M, F, RecvTO)
* block_call(Node, M, F, A, RecvTO
* block_call(Node, M, F, A, RecvTO, SendTO)

Basically the same as Call except for blocking the calling process, and prevent RPC server from handling other requests until current one exits.
* The CallerPid acts as call correlation ID.

Ref: http://www.erlang.org/doc/man/rpc.html

priestjim commented on src/gen_rpc_acceptor.erl:
```diff
@@ -88,6 +88,12 @@ waiting_for_data({data, Data}, #state{socket=Socket,client_node=Node} = State) -
     %% the data
     try erlang:binary_to_term(Data) of
+        {Node, ClientPid, Ref, {block_call, M, F, A}} ->
+            WorkerPid = erlang:spawn(?MODULE, call_worker, [self(), ClientPid, Ref, M, F, A]),
```
Isn't this a "non-blocking" call? The server is supposed to run the function inside the body and reply with the results instead of spawning a worker.

linearregression commented:
Latest check-in addressing comment. 
But I can't figure out how to make TraviscI check in the plt file post file generation, so future projects can just download the pt file instead of relying on travisci. Same logic applies to Erlang plt.
I am less than thrilled of Travici Github deployment provider.
https://docs.travis-ci.com/user/deployment/releases
PS:
* I have a repo that uses travisci to generate plt file for otp & releases
  * TraviscI: https://travis-ci.org/linearregression/dialyxir
  * repo: https://github.com/linearregression/dialyxir
